### PR TITLE
API-2934: fixing HAL links to the notifications

### PR DESCRIPTION
**Pull request by stewabel**
The queue service returns `/notification/{notificationId}` as a notification location. We need to expose this to the client as `/notifications/{notificationId}` so the path is correct for this service, `notifications` being the context of this service

### CODE REVIEW

File: test/unit/uk/gov/hmrc/apinotificationpull/util/XmlBuilderSpec.scala [hunk 1]
```diff
@@ -16,16 +16,25 @@
...
+    when(appContext.apiContext).thenReturn(apiContext)
+    val xmlBuilder = new XmlBuilder(appContext)
+  }
 
   "XmlBuilder.toXml()" should {
 
-    "convert notifications to XML" in {
-      val notifications = Notifications(List("/notifications/123", "/notifications/456"))
+    "convert notifications to XML" in new Setup {
+      val notifications = Notifications(List("/notification/123", "/notification/456"))
```
Comment (gokyo) [inline test/unit/uk/gov/hmrc/apinotificationpull/util/XmlBuilderSpec.scala]: are you sure of this change?

Comment (stewabel) [reply test/unit/uk/gov/hmrc/apinotificationpull/util/XmlBuilderSpec.scala]: yes, the path the queue service returns is `/notification/{notificationId}` we need to expose the path relative to this service by taking the `notificationId` and constructing the path with our api context. The change to this test replicates what we expect the notifications list to look like when retrieved from the queue.

File: test/unit/uk/gov/hmrc/apinotificationpull/util/XmlBuilderSpec.scala [hunk 2]
```diff
@@ -16,16 +16,25 @@
...
+class XmlBuilderSpec extends UnitSpec with MockitoSugar {
+
+  trait Setup {
+    val apiContext = "notifications"
+    val appContext = mock[AppContext]
+    when(appContext.apiContext).thenReturn(apiContext)
+    val xmlBuilder = new XmlBuilder(appContext)
+  }
 
   "XmlBuilder.toXml()" should {
```
Comment (googley42) [inline test/unit/uk/gov/hmrc/apinotificationpull/util/XmlBuilderSpec.scala]: I am confused too. Commit message says
```
The queue service returns /notification/{notificationId} as a notification location. We need to expose this to the client as /notifications/{notificationId} so the path is correct for this service, notifications being the context of this service
```
But code is changing plural `notifications` to `notification`

Comment (stewabel) [reply test/unit/uk/gov/hmrc/apinotificationpull/util/XmlBuilderSpec.scala]: This is the test, the change to from `notifications` to `notification` is to emulate what the queue returns to us. The logic change is then in the `XmlBuilder` which now constructs the path using the `apiContext` from config and the `notificationId` extracted from the path returned by the queue.

File: test/unit/uk/gov/hmrc/apinotificationpull/controllers/NotificationsControllerSpec.scala [hunk 3]
```diff
@@ -98,25 +101,21 @@ class NotificationsControllerSpec extends UnitSpec with WithFakeApplication with
...
 
-      val expectedXml = scala.xml.Utility.trim(
-        <resource href="/notifications/">
-          <link rel="self" href="/notifications/"/>
-          <link rel="notification" href="/notifications/1234"/>
-          <link rel="notification" href="/notifications/6789"/>
-        </resource>
-      )
-
       string2xml(bodyOf(result)) shouldBe expectedXml
```
Comment (googley42) [inline test/unit/uk/gov/hmrc/apinotificationpull/controllers/NotificationsControllerSpec.scala]: I am confues. `notifications` contains 2 elements, but `expectedXml` singular - is this expected?

**Comment by googley42** [comment]
I think `api.context` is missing from `application.conf` ?

**Pull request merged**